components/system/components/PDFViewer.js

```diff
@@ -17,41 +14,63 @@ const STYLES_DOCUMENT = css`
   overflow-y: scroll;
   -ms-overflow-style: none;
   scrollbar-width: none;
-
-  ::-webkit-scrollbar {
-    display: none;
-  }
+  display: grid;
+  place-items: center;
 `;
 
-const STYLES_PAGE = css`
-  margin-bottom: 20px;
+const STYLES_WRAPPER = css`
+  width: calc(100% - 120px);
+  max-width: 850px;
 `;
 
-export default class PDFViewer extends React.Component {
-  state = {
-    numPages: 0,
+const PDFViewer = ({ file, clientId, viewerConfig, ...rest }) => {
+  viewerConfig = {
+    showAnnotationTools: false,
+    enableFormFilling: false,
+    showLeftHandPanel: false,
+    showDownloadPDF: false,
+    showPrintPDF: false,
+    showPageControls: false,
+    dockPageControls: false,
+    embedMode: "IN_LINE",
   };
 
-  onDocumentLoadSuccess({ numPages }) {
-    this.setState({
-      numPages,
+  const initializeViewer = () => {
+    let adobeDCView = new AdobeDC.View({
+      clientId,
+      divId: "adobe-dc-view",
     });
-  }
-
-  render() {
-    const { numPages } = this.state;
-
-    return (
-      <Document
-        file={this.props.file}
-        onLoadSuccess={this.onDocumentLoadSuccess.bind(this)}
-        {...this.props}
-        css={STYLES_DOCUMENT}
-      >
-        {Array.from(new Array(numPages), (el, index) => (
-          <Page key={`page_${index + 1}`} pageNumber={index + 1} css={STYLES_PAGE} />
-        ))}
-      </Document>
+
+    adobeDCView.previewFile(
+      {
+        content: {
+          location: {
+            url: file,
+          },
+        },
+        metaData: { fileName: "title" },
+      },
+      viewerConfig
     );
-  }
-}
+  };
+
+  React.useEffect(() => {
+    document.addEventListener("adobe_dc_view_sdk.ready", initializeViewer);
+  });
+
+  return (
+    <>
+      <Head>
+        <script src="https://documentcloud.adobe.com/view-sdk/main.js" />
+      </Head>
```

Comment:
Worth double checking if this will this insert the script every time this component is rendered/mounted. I would typically recommend using a HOC (such as [react-async-script](https://github.com/dozoisch/react-async-script)) to fetch any external dependencies so that the lifecycle is managed better.

Reply:
Okay. I'm gonna look into this